File: solutions/dart/bob/1/lib/bob.dart
class Bob {
  String response(String question) {
    question = question.trimRight();
    // response to silence
    if (question == "") {
      return "Fine. Be that way!";
    }
    // response to yelling (must contain some text)
    if (question.contains(RegExp(r'[a-zA-Z]')) &&
        question == question.toUpperCase()) {
      if (question.endsWith("?")) {
        return "Calm down, I know what I'm doing!";
      } else {
        return "Whoa, chill out!";
      }
    }
    // response to normal question
    if (question.endsWith("?")) {
      return "Sure.";
    }
    // default response
    return "Whatever.";
  }
}

File: solutions/swift/bob/1/Sources/Bob/Bob.swift
import Foundation

class Bob {
  static func response(_ message: String) -> String {

    let trimmedMessage = message.trimmingCharacters(in: .whitespacesAndNewlines)
    if allCaps(trimmedMessage) {
      if trimmedMessage.hasSuffix("?") {
        return "Calm down, I know what I'm doing!"
      } else {
        return "Whoa, chill out!"
      }
    } else if trimmedMessage.hasSuffix("?") {
      return "Sure."
    } else if trimmedMessage == "" {
      return "Fine. Be that way!"
    } else {
      return "Whatever."
    }
  }
}

func allCaps(_ message: String) -> Bool {
  return message.rangeOfCharacter(from: .letters) != nil && message == message.uppercased()
}
